docs/experiments/project-depends-on.md

```diff
@@ -0,0 +1,35 @@
+---
+slug: project-depends-on
+append_help_link: true
+description: "project-depends-on lets Semgrep rules only returns results if the project depends on a specific version of a third-party package."
+---
+
+# project-depends-on
+
+Under this key, NPM or PyPI dependencies can be specified along with the semver range that the rule should trigger for. `project-depends-on` filters the rule unless one of the children is matched by a lockfile. In this initial release, the key is named `r2c-internal-project-depends-on` to signal that the syntax & behavior for the key is not stable and may be subject to removal or future changes. 
```

Comment:
I really like this `r2c-internal-` prefix. Do you think it’s worth highlighting through an admonition?